Add GrabPay support

cheeliang-stripe opened the pull request:
### Summary & motivation

Add typing support for GrabPay flows.

### Testing & documentation

Added automated tests for the new types.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/stripe/stripe-js?pullRequest=118) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/stripe/stripe-js?pullRequest=118) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/stripe/stripe-js?pullRequest=118) it.</sub>

christopher-stripe commented on types/api/PaymentMethods.d.ts:
```diff
@@ -33,6 +33,8 @@ declare module '@stripe/stripe-js' {
 
     fpx?: PaymentMethod.Fpx;
 
+    grabpab?: PaymentMethod.GrabPay;
```
```suggestion
    grabpay?: PaymentMethod.GrabPay;
```

cheeliang-stripe replied:
🙈  thank you!

christopher-stripe commented on types/stripe-js/payment-intents.d.ts:
```diff
@@ -118,6 +118,15 @@ declare module '@stripe/stripe-js' {
     };
   }
 
+  interface CreatePaymentMethodGrabPayData extends PaymentMethodCreateParams {
```
Can a `grabpay` PaymentMethod be created directly? i.e. without going through a PaymentIntent.

The API ref makes it seem like that is the case, though I know sometimes the API ref is misleading. 

In Stripe.js land, that means that someone could also write

```js
stripe.createPaymentMethod({
  type: 'grabpay',
  grabpay: {},
})
```
If this is a legitimate use case, the `CreatePaymentMethodGrabPayData` needs to be added to the `CreatePaymentMethodData` enum above.

cheeliang-stripe replied:
A `grabpay` PM doesn't hold any data, but we still do allow users to create one directly. Fixed.

christopher-stripe reviewed (changes requested):
Looks good! Just one quick typo and a question

cheeliang-stripe commented:
@christopher-stripe: I've addressed the issues that you've pointed out.
I've also rearranged my changes to keep them sorted alphabetically between other payment methods.
Appreciate another review!

christopher-stripe commented on types/stripe-js/payment-intents.d.ts:
```diff
@@ -90,6 +91,15 @@ declare module '@stripe/stripe-js' {
+    /**
+     * Can be omitted as there are no GrabPay-specific fields.
+     */
+    grabpay: {};
```
Is this field optional? Or is it required but can be blank?
```suggestion
    grabpay?: {};
```

cheeliang-stripe replied:
Fixed!

christopher-stripe commented on tests/types/index.ts:
```diff
@@ -468,6 +468,16 @@ stripe.confirmGiropayPayment('', {payment_method: ''}, {handleActions: false});
+
+stripe.confirmGrabPayPayment('', {payment_method: ''});
+
+stripe.confirmGrabPayPayment('', {payment_method: ''}, {handleActions: false});
```
Could you also add some type tests for `stripe.createPaymentMethod` with a GrabPay?

cheeliang-stripe replied:
Added!

christopher-stripe reviewed (commented):
Two quick comments but otherwise looks good!

cheeliang-stripe commented:
@christopher-stripe: I've addressed your comments.

christopher-stripe reviewed (approved):
Awesome, thank you!
